### PR TITLE
Nuget packages updates and more telemetry for payments and invoices

### PR DESCRIPTION
**Pull request by tiagonmas**

### CODE REVIEW

File: BancoAlimentar.AlimentaEstaIdeia.Web/Api/EasyPayGenericNotification.cs [hunk 1]
```diff
@@ -59,6 +61,12 @@ public IActionResult PostAsync(GenericNotificationRequest notificationRequest)
             this.HttpContext.Items.Add(KeyNames.GenericNotificationKey, notificationRequest);
             if (notificationRequest != null)
             {
+                this.telemetryClient.TrackEvent("easypay-generic", new Dictionary<string, string>()
+                {
+                    { "notificationRequestType", notificationRequest.Type.ToString() },
+                    { "notificationRequestId", notificationRequest.Id.ToString() },
+                    { "notificationRequestKey", notificationRequest.Key.ToString() },
+                });
```
Comment (guerrerotook) [inline BancoAlimentar.AlimentaEstaIdeia.Web/Api/EasyPayGenericNotification.cs:69]: That telemetry is already included. 

image

File: BancoAlimentar.AlimentaEstaIdeia.Web/Api/EasyPayPaymentNotification.cs [hunk 2]
```diff
@@ -55,6 +60,15 @@ public async Task<IActionResult> PostAsync(TransactionNotificationRequest value)
...
             {
+                this.telemetryClient.TrackEvent("easypay-payment", new Dictionary<string, string>()
+                    {
+                        { "Method", value.Method },
+                        { "PublicDonationId", value.Id.ToString() },
+                        { "TransactionKey", value.Transaction.Key },
+                        { "TransactionId", value.Transaction.Id.ToString() },
+                        { "Paid", value.Transaction.Values.Paid.ToString() },
+                    });
+
```
Comment (guerrerotook) [inline BancoAlimentar.AlimentaEstaIdeia.Web/Api/EasyPayPaymentNotification.cs:71]: Same here, we already have the information about the easypay data as a custom properties

**Review by guerrerotook — CHANGES_REQUESTED** [review]
Look at the already custom properties that we are sending here. 

If you want to add more please do, https://github.com/banco-alimentar/alimentestaideia.pt/blob/d1fb6f6ca7beb05f6dc4fac3f2b5d0ad38a2080a/BancoAlimentar.AlimentaEstaIdeia.Web/Telemetry/HttpTelemetryInitializer.cs#L111 But remove those already included.

File: BancoAlimentar.AlimentaEstaIdeia.Web/Api/EasyPayPaymentNotification.cs (hunk 2 above)
Comment (tiagonmas) [reply BancoAlimentar.AlimentaEstaIdeia.Web/Api/EasyPayPaymentNotification.cs:71]: ok. I'll remove. I missed seeing them in App insights. Guess I was not looking correctly :)

File: BancoAlimentar.AlimentaEstaIdeia.Web/Api/EasyPayGenericNotification.cs (hunk 1 above)
Comment (tiagonmas) [reply BancoAlimentar.AlimentaEstaIdeia.Web/Api/EasyPayGenericNotification.cs]: ok. just removed it in f9f6824f5544eb2271ae9adb7dad9a135bab56a2

File: BancoAlimentar.AlimentaEstaIdeia.Web/Api/EasyPayPaymentNotification.cs (hunk 2 above)
Comment (tiagonmas) [reply BancoAlimentar.AlimentaEstaIdeia.Web/Api/EasyPayPaymentNotification.cs]: done in f9f6824f5544eb2271ae9adb7dad9a135bab56a2

**Review by guerrerotook — APPROVED** [review]

**Pull request merged**